wagtail/images/models.py

```diff
@@ -509,14 +499,220 @@ def create_rendition(self, filter: "Filter") -> "AbstractRendition":
+                        # skip to the next filter
+                        break
+        else:
+            # Renditions are not prefetched, so attempt to find suitable
+            # items in the cache or database
+
+            # Query the cache first (if enabled)
+            if self.renditions_cache is not None:
+                cache_keys = [
+                    Rendition.construct_cache_key(
+                        self.id, filter.get_cache_key(self), spec
+                    )
+                    for spec, filter in filters_by_spec.items()
+                ]
+                for rendition in self.renditions_cache.get_many(cache_keys).values():
+                    filter = filters_by_spec[rendition.filter_spec]
+                    found[filter] = rendition
+
+            # For items not found in the cache, look in the database
+            not_found = tuple(f for f in filters if f not in found)
+            if not_found:
+                lookup_q = Q()
+                for filter in not_found:
+                    lookup_q |= Q(
+                        filter_spec=filter.spec,
+                        focal_point_key=filter.get_cache_key(self),
+                    )
+                for rendition in self.renditions.filter(lookup_q):
+                    filter = filters_by_spec[rendition.filter_spec]
+                    found[filter] = rendition
+        return found
+
+    def create_renditions(
+        self, *filters: "Filter"
+    ) -> Dict["Filter", "AbstractRendition"]:
+        """
+        Creates multiple ``Rendition`` instances with image files reflecting the supplied
+        ``filters``, and returns them as a ``dict`` keyed by the relevant ``Filter`` instance.
+        Where suitable renditions already exist in the database, they will be returned instead,
+        so as not to create duplicates.
+
+        This method is usually called by ``Image.get_renditions()``, after first
+        checking that a suitable rendition does not already exist.
+
+        Note: If using custom image models, an instance of the custom rendition
+        model will be returned.
+        """
+        Rendition = self.get_rendition_model()
+
+        if not filters:
+            return {}
+
+        if len(filters) == 1:
+            # create_rendition() is better for single renditions, as it can
+            # utilize QuerySet.get_or_create(), which has better handling of
+            # race conditions
+            filter = filters[0]
+            return {filter: self.create_rendition(filter)}
+
+        return_value: Dict[Filter, AbstractRendition] = {}
+        filter_map: Dict[str, Filter] = {f.spec: f for f in filters}
+
+        with self.open_file() as file:
+            original_image_bytes = file.read()
+
+        to_create = []
+
+        def _generate_single_rendition(filter):
+            image_file = self.generate_rendition_file(
+                filter, source=ContentFile(original_image_bytes, name=self.file.name)
```

Comment:
Question: Can we not just use `source=self.file` here? That way Django can deal with the file buffering itself

Reply:
So, I've been testing with https://github.com/wagtail/bakerydemo/pull/367
if we change to `source=self.file` I had to reload a number of times until all renditions were generated as oppposed to all in one go with this implementation